azext_edge/edge/params.py

```diff
@@ -302,6 +310,822 @@ def load_iotops_arguments(self, _):
+            type=int,
+        )
+        context.argument(
+            "retain",
+            options_list=["--retain"],
+            arg_type=get_enum_type(MqttRetainType, default=MqttRetainType.KEEP.value),
+            help="Retain setting to specify whether the data flow should keep the retain "
+            "flag on MQTT messages. Setting this ensures whether or not the remote broker "
+            "has the same messages retained as the local broker.",
+        )
+        context.argument(
+            "session_expiry",
+            options_list=["--session-expiry"],
+            help="The session expiry interval in seconds for the data flow MQTT client. "
+            "Min value: 0.",
+            type=int,
+        )
+
+    with self.argument_context("iot ops dataflow endpoint create adx") as context:
+        context.argument(
+            "database_name",
+            options_list=["--database-name", "--db"],
+            help="The name of the Azure Data Explorer database.",
+        )
+        context.argument(
+            "host",
+            options_list=["--host"],
+            help="Host of the Azure Data Explorer is "
+            "Azure Data Explorer cluster URI. In the form "
+            "of https://cluster.region.kusto.windows.net",
+        )
+        context.argument(
+            "authentication_type",
+            options_list=["--auth-type"],
+            choices=CaseInsensitiveList(
+                [
+                    DataflowEndpointAuthenticationType.SYSTEMASSIGNED.value,
+                    DataflowEndpointAuthenticationType.USERASSIGNED.value,
+                ]
+            ),
+        )
+
+    with self.argument_context("iot ops dataflow endpoint create adls") as context:
+        context.argument(
+            "secret_name",
+            options_list=["--secret-name", "--sn"],
+            help="The name for the kubernetes secret that contains SAS token.",
+            arg_group="Access Token",
+        )
+        context.argument(
+            "storage_account_name",
+            options_list=["--storage-account-name", "--san"],
+            help="The name of Azure Data Lake Storage Gen2 account.",
+        )
+        context.argument(
+            "authentication_type",
+            options_list=["--auth-type"],
+            choices=CaseInsensitiveList(
+                [
+                    DataflowEndpointAuthenticationType.SYSTEMASSIGNED.value,
+                    DataflowEndpointAuthenticationType.USERASSIGNED.value,
+                    DataflowEndpointAuthenticationType.ACCESSTOKEN.value,
+                ]
+            ),
+        )
+
+    with self.argument_context("iot ops dataflow endpoint create fabric-onelake") as context:
+        context.argument(
+            "workspace_name",
+            options_list=["--workspace-name", "--wn"],
+            help="The Microsoft Fabric workspace name. Note: The default 'my workspace' isn't supported.",
```

Comment:
the note is exteremly confusing - if the default doesnt work then why not have a different default? Or just make this a required param?

Reply:
this is from the official ms learning documentation, why is this confusing? my workspace is always there when you sign in  Microsoft Fabric workspace

Reply:
<img width="727" alt="image" src="https://github.com/user-attachments/assets/84a4b43f-c666-439b-ab16-8ae6e797f4f0" />